src/extension.ts

```diff
@@ -44,6 +45,18 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         }
     };
 
+    const workspaceFolder = vscode.workspace.workspaceFolders?.[0];
+
+    if (workspaceFolder) {
+        const fullPath = `${workspaceFolder.uri.fsPath}/pyproject.toml`;
+        watcher = vscode.workspace.createFileSystemWatcher(fullPath);
+
+        watcher.onDidChange((uri) => {
+            console.log('pyproject.toml changed');
+            vscode.commands.executeCommand('mypy-type-checker.restart', uri.fsPath);
```

Comment:
You can directly call the restart method that is available in this context.
```suggestion
            await runServer();
```

Reply:
Thanks for suggestions.  please check new commit.